permissions/src/androidTest/java/com/google/accompanist/permissions/TestUtils.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.permissions
+
+import android.app.Instrumentation
+import android.os.Build
+import androidx.activity.ComponentActivity
+import androidx.compose.ui.test.junit4.AndroidComposeTestRule
+import androidx.lifecycle.Lifecycle
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+
+internal fun <T : ComponentActivity> simulateAppComingFromTheBackground(
+    composeTestRule: AndroidComposeTestRule<ActivityScenarioRule<T>, T>
+) {
+    // Recreate Activity and set content again to check for the permission again
+    composeTestRule.activityRule.scenario.moveToState(Lifecycle.State.STARTED)
+    // A weird dialog appears when moving to the STARTED state
+    UiDevice.getInstance(InstrumentationRegistry.getInstrumentation()).pressBack()
+}
+
+internal fun grantPermissionProgrammatically(
```

Comment:
Have you thought about using [GrantPermissionRule](https://developer.android.com/reference/androidx/test/rule/GrantPermissionRule)?

Reply:
I use that in other tests that don't require changing the status of that permission within the test itself. But I need this for some others